Android/src/org/droidplanner/android/fragments/mode/ModeAutoFragment.java

```diff
@@ -1,17 +1,278 @@
+                case AttributeEvent.MISSION_ITEM_UPDATED:
+                    mission = drone.getAttribute(AttributeType.MISSION);
+                    int currentWaypoint = intent.getIntExtra(AttributeEventExtra.EXTRA_MISSION_CURRENT_WAYPOINT, 0);
+                    mission.setCurrentMissionItem(currentWaypoint);
+                case AttributeEvent.GPS_POSITION:
+                    updateMission();
+                    break;
+
+                case AttributeEvent.PARAMETER_RECEIVED:
+                    String paramName = intent.getStringExtra(AttributeEventExtra.EXTRA_PARAMETER_NAME);
+                    if(paramName.equals(WPNAV_SPEED)) {
+                        double value = intent.getDoubleExtra(AttributeEventExtra.EXTRA_PARAMETER_VALUE, 2000.0);
+                        updateWaypointSpeed(value);
+                    }
+                    break;
+
+                case AttributeEvent.AUTOPILOT_MESSAGE:
+                    String message = intent.getStringExtra(AttributeEventExtra.EXTRA_AUTOPILOT_MESSAGE);
+                    Timber.i(message);
+            }
+        }
+    };
+    private Mission mission;
+
+    private TextView waypointSpeed;
+    private ProgressBar missionProgress;
 
-public class ModeAutoFragment extends Fragment {
 
 	@Override
 	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 		return inflater.inflate(R.layout.fragment_mode_auto, container, false);
 	}
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        view.findViewById(R.id.mc_pause).setOnClickListener(this);
+        view.findViewById(R.id.mc_restart).setOnClickListener(this);
+        view.findViewById(R.id.mc_next).setOnClickListener(this);
+        view.findViewById(R.id.mc_prev).setOnClickListener(this);
+        view.findViewById(R.id.wp_speed_plus).setOnClickListener(this);
+        view.findViewById(R.id.wp_speed_minus).setOnClickListener(this);
+        waypointSpeed = (TextView) view.findViewById(R.id.wp_speed);
+        missionProgress = (ProgressBar) view.findViewById(R.id.mission_progress);
+    }
+
+    @Override
+	public void onClick(View v) {
+		Parameters params = drone.getAttribute(AttributeType.PARAMETERS);
+        Parameter speed = params.getParameter(WPNAV_SPEED);
+        if(mission == null){
+            mission = drone.getAttribute(AttributeType.MISSION);
+        }
+		switch(v.getId()){
+			case R.id.mc_pause: {
+                drone.pauseAtCurrentLocation();
+                break;
+            }
+			case R.id.mc_restart: {
+                if(mission.getCurrentMissionItem() == mission.getMissionItems().size()){
+                    resetAutoMode();
+                }
+                msg_mission_set_current msg = new msg_mission_set_current();
+                msg.seq = 0;
+                ExperimentalApi.sendMavlinkMessage(drone, new MavlinkMessageWrapper(msg));
```

Comment:
This command doesn't work yet.  See diydrones/ardupilot#2584.  It isn't on the list to be fixed with 3.4.  So maybe we should remove this button for now, or find a workaround.